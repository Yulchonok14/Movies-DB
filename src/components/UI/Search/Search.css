.Search {
    display: flex;
    justify-content: space-between;
    align-items: center;
    box-sizing: border-box;
    flex-direction: column;
}

.Search input {
    color: #FFFFFF;
    background-color: #232323;
    width: 100%;
    height: 60px;
    border: none;
    text-transform: capitalize;
    font-size: 20px;
    padding-left: 20px;
    margin: 10px 0;
    outline: none;
    opacity: 0.8;
}

.Search input::placeholder {
    color: #555555;
}

.Search button {
    width: 230px;
    color: #FFFFFF;
    background-color: #F65261;
    border-radius: 4px;
    text-transform: uppercase;
    height: 60px;
    font-size: 20px;
    border: none;
    outline: none;
}

.Search button:hover {
    box-shadow: 9px 1px 11px rgba(192,192,192,192.85), 8px 10px 10px rgba(192,192,192,192.82);
}

@media (min-width: 600px) {
    .Search {
        flex-direction: row;
    }

    .Search input {
        width: 80%;
    }
}