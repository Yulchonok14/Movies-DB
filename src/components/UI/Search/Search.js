import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import classes from './Search.css';

const Search = (props) => {
    const inputRef = useRef(null);
    return (
        <div className={classes.Search}>
            <input ref={inputRef} placeholder="search" defaultValue={props.searchText}/>
            <button onClick={() => props.clicked(inputRef.current.value.toLowerCase())}>search</button>
        </div>
    )
};

Search.propTypes = {
    clicked: PropTypes.func.isRequired
};

export default Search;