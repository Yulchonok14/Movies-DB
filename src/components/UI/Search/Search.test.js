import React, { useRef } from 'react';

import { configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Search from './Search';

configure({
    adapter: new Adapter()
});

jest.mock('react', () => {
    const originReact = jest.requireActual('react');
    const mUseRef = jest.fn();
    return {
        ...originReact,
        useRef: mUseRef
    };
});

describe('<Search/>', () => {
    const mockCallBack = jest.fn();
    let wrapper;

    beforeEach(() => {
        const mRef = { current: { value: 'Any value' } };
        useRef.mockReturnValueOnce(mRef);
        wrapper = shallow(<Search clicked={mockCallBack}/>);
    });

    it('Search renders correctly', () => {
        const tree = renderer
            .create(<Search clicked={mockCallBack}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Upon the button click callback should be called', () => {
        wrapper.find('button').simulate('click');
        expect(mockCallBack).toHaveBeenCalledWith('any value');
    });
});