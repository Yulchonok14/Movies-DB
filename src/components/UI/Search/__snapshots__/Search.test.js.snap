// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> Search renders correctly 1`] = `
<div>
  <input
    placeholder="search"
  />
  <button
    onClick={[Function]}
  >
    search
  </button>
</div>
`;
